go/host/l1/publisher.go

```diff
@@ -320,72 +338,112 @@ func (p *Publisher) ResyncImportantContracts() error {
 // this method is guarded by a lock to ensure that only one transaction is attempted at a time to avoid nonce conflicts
 // todo (@matt) this method should take a context so we can try to cancel if the tx is no longer required
 func (p *Publisher) publishTransaction(tx types.TxData) error {
-	// this log message seems superfluous but is useful to debug deadlock issues, we expect 'Host issuing l1 tx' soon
-	// after unless we're stuck blocking.
-	p.logger.Info("Host preparing to issue L1 tx")
-
 	p.sendingLock.Lock()
 	defer p.sendingLock.Unlock()
 
-	retries := -1
-
 	nonce, err := p.ethClient.Nonce(p.hostWallet.Address())
 	if err != nil {
 		return fmt.Errorf("could not get nonce for L1 tx: %w", err)
 	}
 
-	// while the publisher service is still alive we keep trying to get the transaction into the L1
-	for !p.hostStopper.IsStopping() {
-		retries++ // count each attempt so we can increase gas price
-
-		// update the tx gas price before each attempt
-		tx, err := ethadapter.SetTxGasPrice(p.sendingContext, p.ethClient, tx, p.hostWallet.Address(), nonce, retries)
-		if err != nil {
-			return errors.Wrap(err, "could not estimate gas/gas price for L1 tx")
+	// Try initial publish
+	if err := p.executeTransaction(tx, nonce, 0); err != nil {
```

Comment:
Can't we invoke ``publishXWithRetry`` directly, and it will start at 0?

Reply:
good spot